presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SingleDistinctOptimizer.java

```diff
@@ -81,6 +81,9 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<S
             if (masks.size() != 1 || node.getMasks().size() != node.getAggregations().size()) {
                 return context.defaultRewrite(node, Optional.empty());
             }
+            if (!Iterables.getOnlyElement(masks).getName().contains("distinct")) {
```

Comment:
I made a small change to check if the aggregation is filter instead of hard coded string.